daos/authors.go

```diff
@@ -58,6 +58,6 @@ func FetchAllAuthors(ctx context.Context) (models.AuthorSlice, error) {
 
 // FetchAuthorByID returns the Author associated with the id
 func FetchAuthorByID(ID int, ctx context.Context) (*models.Author, error) {
-	contextExecutor := getContextExecutor(nil)
+	contextExecutor := getContextExecutor(&sql.Tx{})
```

Comment:
Why are we not passing nil here?

Reply:
added them to verify some sql.Tx properties, reverted back to nil